src/encoding.ts

```diff
@@ -61,16 +61,17 @@ export function encodeQueryValue(input: QueryValue): string {
       .replace(AMPERSAND_RE, "%26")
       .replace(ENC_BACKTICK_RE, "`")
       .replace(ENC_CARET_RE, "^")
+      .replace(EQUAL_RE, "%3D")
   );
 }
 
 /**
- * Like `encodeQueryValue` but also encodes the `=` character.
+ * Same as `encodeQueryValue`
  *
  * @param text - string to encode
  */
 export function encodeQueryKey(text: string | number): string {
-  return encodeQueryValue(text).replace(EQUAL_RE, "%3D");
+  return encodeQueryValue(text);
```

Comment:
/cc @posva was there any specific reason `=` wasn't being encoded before? (This was taken from vue-router utils i guess)

Reply:
Indeed it is parsable.. I think it would be nice if we could preserve a consistent behavior between ufo, vue-router (and `URL`) spec if you are okay I can make a PR to `vue-router` as well to keep them in sync and avoid mistmatching behavior between different places (as we use ufo in nuxt)

Reply:
I will give it a look in 2-3 weeks. I did a lot of testing in different browsers when writing that code and I imagine that if I added 2 functions instead of one, there must be a reason 😆

Reply:
Here is the discrepancy:

- When directly entering some characters in a URL, the browser encodes some of them. The `=` within a query value, isn't encoded e.g. reads as `=` in `location.href`.
- `URLSearchParams()` encodes `=`

Depending on the browsers, they display the value on the URL as encoded or not, therefore I tried, as much as possible, to display the original characters across browsers when they didn't need encoding. Also, Vue Router supports some browsers that do not have full `URLSearchParams()` support.

BTW I don't think the URL spec specifies that `=` must be encoded in query values.

When I did the research on what was supported and what the spec said, it turned out `encodeURIComponent()` was doing too much in some scenarios and that's why it's not used in vue router. Unless there is a bug, changing this behavior in vue router would be tiny breaking change, so it's better to keep it the way it is, being more readable across browsers.

Cheers

Reply:
Thanks for detailed feedback. Is there a list of officially supported browsers by vue-router btw? And do you think it would be an issue if ufo (in a major version) choose to always encode things like `=` with `URLSearchParams` compatibility? (basically: does vue-router guarantees to decode encoded values like `=` in query even if it doesn't encodes it?)

Reply:
We don't have a list, it's the same as Vue 3 😅 

It's better to try by yourself, there might be problems with redirects and reloads. However, if I were you, since there is currently no problem with the encoding, I wouldn't change it in ufo either. The opened issue is fundamentally wrong, specifically, this statement is not true:

> When we encode a query not a url. we should use [encodeURIComponent](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURI) not [encodeURI](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURI#encodeuri_vs._encodeuricomponent).

The spec specifies encode sets, not functions to use. Since the issue lacks a real-world scenario with a problem, I would personally wait to see if this fixes anything, if not, it's better to keep the current behavior for the reasons I mentioned in my previous comment

Reply:
The thing is, for the next major version of UFO I am thinking of relying on native `URL` and `URLSearch` params for encoding and decoding since it has good coverage in platforms (I guess at least all Vue 3 supported targets -- have to double check) and as a result, we might end up with compatibility issues if didn't anticipate edge cases.

Thanks for feedbacks anyway, if we want to do this, i will certainly reach you out for more through tests across ecosystem packages.

Reply:
FYI I wanted to rely on URL and URLSearchParams but there were small bugs in some old versions of browsers that forced me to not use them yet. I do hope to use them in the future major version of Vue Router though!